src/main/src/main/java/org/geoserver/catalog/ResourcePool.java

```diff
@@ -2343,35 +2345,38 @@ private void flushState(ContentDataStore contentDataStore, String nativeName)
     }
     
     public DataStoreInfo clone(final DataStoreInfo source, boolean allowEnvParametrization) {
-        DataStoreInfo target = catalog.getFactory().createDataStore();
-        target.setEnabled(source.isEnabled());
-        target.setName(source.getName());
-        target.setDescription(source.getDescription());
-        target.setWorkspace(source.getWorkspace());
-        target.setType(source.getType());
+        DataStoreInfo target;
+        try {
+            target = (DataStoreInfo) SerializationUtils.clone(source);
```

Comment:
I guess this is to get out of breaking the code each time a new field is added?

I am used to using copy constructors to hide individual fields being set ...